Added a number of 3D models of coaxial connectors

Misca1234 opened the pull request:
Added a number 3D models of coaxial connector

All models are script generated

3D model push

3D model script push

![bild](https://user-images.githubusercontent.com/25547797/50575585-7c9e0e00-0e01-11e9-891f-b74f5bccd71e.png)

![bild](https://user-images.githubusercontent.com/25547797/50575587-845db280-0e01-11e9-9552-95ca6edc4e7c.png)

![bild](https://user-images.githubusercontent.com/25547797/50575590-89226680-0e01-11e9-9006-f30361395d73.png)

![bild](https://user-images.githubusercontent.com/25547797/50575592-8e7fb100-0e01-11e9-99f3-9baa3b47d2d1.png)

![bild](https://user-images.githubusercontent.com/25547797/50575595-93446500-0e01-11e9-93c7-8e54e18f62f4.png)

![bild](https://user-images.githubusercontent.com/25547797/50575597-98a1af80-0e01-11e9-92ef-bcedf2b1c3b8.png)

![bild](https://user-images.githubusercontent.com/25547797/50575599-9d666380-0e01-11e9-94a1-352d6232d3a4.png)




------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items as you complete them:

- [ ] A screenshot of the model(s) aligned with KiCad footprints is *very useful*
- [ ] Was the 3D model(s) created in a mechanical modelling program?
- [ ] Source files (or scripts) have been contributed 
- [ ] Step model submitted in 1:1 scale
- [ ] WRL model submitted in 1:2.54 scale
- [ ] 3D model alignment checked against KiCad footprint
- [ ] WRL model uses correct material properties
- [ ] Geometry check performed against mechanical models

Shackmeister commented:
These looks very nice!
Hope we can get them finished for 5.1 :)

I can imagine these are pretty hard to script, and I'm a little unsure about how pedantic I should be. I think the external sizes should be correct, and the internal should at least looks right, if I'm being silly with details, let me know.

It's kinda hard to describe the dimension issues, since the models are quite complex.
I have added a zip with some Freecad files with the original model and your scripted model, this should hopefully make it easier to check the differences

General notes
- Manufacturers typically use chamfers on pins, where as you use fillets.
- Full name is not inside step file

BNC_TEConnectivity_1478204:

- Dimensions seems off, please see attached FCstd

SMA_Amphenol_132134_10:

- Internal plastic piece should go all the way out to the "brass"
- Cylinder seems too big.
- Threads are slightly off in size, but that might be very very tricky to get right.
- hole for plug, is too small,link below shows sizes for SMA connectors quite well.
http://www.honorconnector.com/sma-connectors/rp-sma-connectors.html


SMA_Amphenol_132134-11:

- see  notes above
- Feet seems a bit weird
![sma connector](https://user-images.githubusercontent.com/5279723/52891905-45f03d00-318e-11e9-94ef-77d0fa8e2d8f.png)

SMA_Amphenol_132134-14:

- see notes above
- center seems to have an error
![center_pin](https://user-images.githubusercontent.com/5279723/52891987-f52d1400-318e-11e9-9bf5-3f7c237c1d24.png)

SMA_Amphenol_132134-16:

- see notes above

SMA_Amphenol_132134:

- See notes above
- measurements seems off here and there
- Connector is not a single piece, 
![sma_amphenol_132134](https://user-images.githubusercontent.com/5279723/52892204-aaac9700-3190-11e9-8587-f25340f4f1bd.png)

SMA_Amphenol_132289_EdgeMount:

- See notes above:
- This shows 2 parts, and has a naming issue.
- general dimensions seems off, see attached zip file

SMA_Amphenol_901-144_Vertical:

- See notes above
- not properly fused.
- This part doesnt have the "extra seating" for the legs.

SMA_Molex_73251-1153_EdgeMount_Horizontal:

- See above
- not properly fused, there is a super tiny cylinder.
- Center pin is too thick
- Base is too high. Check FCstd file for more.

SMA_Molex_73251-2200_Horizontal:

- See above
- size slightly off, see FCstd

SMA_Samtec_SMA-J-P-H-ST-EM1_EdgeMount:
- see above
- size slightly off, see FCstd

U.FL_Molex_MCRF_73412-0110_Vertical:
- adding the cut in on the cylinder would be an awesome detail
- pins seems inconsistent with original model, see FCstd
[Coaxial_connector_check.zip](https://github.com/KiCad/kicad-packages3D/files/2871506/Coaxial_connector_check.zip)

Misca1234 commented:
I was unsure how to do the thread but then I got a tip that the thread could be made with slightly tilted cylinders and then it was fairly easy

I go through it tomorrow

poeschlr commented:
I would not include a too detailed model for threads. At least not in the step version. (In MCAD threads are typically communicated with attributes instead of modeling them. This is done to reduce the complexity of the rendering and the model. It is also done to easily allow making dimensioned drawings as an attribute is easier to parse then a modeled thread.)

My suggestion therefore is to have basically two models in the script. Simple one for step (with nominal diameter. Freecad sadly can not add attributes for threads right now. I am also uncertain if step even includes options to communicate this info between different tools or if this is tool specific.)
And the more complex one for wrl. (The wrl model then really only needs to look kind of nice. No need to make the thread fully compliant to a true thread form.)

Shackmeister commented:
@poeschlr My point was not to have an actual thread, revolved triangled (as used) are much smaller in size and complexity. The external diameter should however be right since they are often used as reference for a hole in a box or panel.

Misca1234 commented:
> Manufacturers typically use chamfers on pins, where as you use fillets.
I can live with that

>  Full name is not inside step file
Then something is wrong in the scripts call addLicenseToStep, not checking now what it is 


BNC_TEConnectivity_1478204
Height have bene increased

SMA_Amphenol_132134_10
Fixed Internal plastic piece
Fixed hole for plug
Made the thread slightly larger


SMA_Amphenol_132134-11
as SMA_Amphenol_132134_10

SMA_Amphenol_132134-14
as SMA_Amphenol_132134_10
Cant see the error in the top

SMA_Amphenol_132134-16
as SMA_Amphenol_132134_10


SMA_Amphenol_132134
as SMA_Amphenol_132134_10
Connected the pieces



SMA_Amphenol_132289
as SMA_Amphenol_132134_10
> This shows 2 parts, and has a naming issue.
Naming issus ?


SMA_Amphenol_901-144_Vertical
as SMA_Amphenol_132134_10


SMA_Amphenol_901_144
as SMA_Amphenol_132134_10
Removed seating

U_FL_Molex_MCRF_73412_0110
> adding the cut in on the cylinder would be an awesome detail
Yes, but not this round
fixed the pin sizes


let me know if you feel that the sizes also should be changed

antoniovazquezblanco commented:
Taking the liberty o assing @Shackmeister as has already almost done all the review.

Is this ok with you?

Misca1234 commented:
I have added a union test for the coaxial connector in the scripts, see push
https://github.com/easyw/kicad-3d-models-in-freecad/pull/308

I have corrected those 3D models that was not correctly unioned and repushed the models to this push

Franck78 commented:
```
                if t == 'round':
                    case1 = cq.Workplane("XY").workplane(offset=1.0).moveTo(0.0, 0.0).circle(w / 2.0, False).extrude(0.0 - (ld + 1.0))
                    case1 = case1.faces("<Z").fillet(w / 2.2)
                elif t == 'rect':

```

The strange pin termination is a bug somewhere **in** the call fillet( w / 2.2)

It affects **only** the step.

I have tried different size for fillet size, nothing works on the small diameter of that pin. Joel will translate *pinaillage* , not a stopper for the Amphenol images.

that's the termination for the rounded pin : perfect (wrml)

![amp132134-14wrl-ok](https://user-images.githubusercontent.com/1778884/94190189-1518d380-feac-11ea-81e6-555f8b7bd190.png)

**73412**

"let me know if you feel that the sizes also should be changed"
what size? It is ok as it is.
![sma73412](https://user-images.githubusercontent.com/1778884/94193971-52cc2b00-feb1-11ea-9041-a5ef7e344b30.png)


~10 3dmodels go to go

Franck78 commented:
@easyw 
just a ping for Easy. Do you know of a limitation in the fillet function applied on an extruded circle face ?
Here, with small diameter, the fillet is buggy (Shackmeister's images).
I have exported a basic 0.5mm tube with 0.3 fillet and the .step is correct.
